dtsync/message.go

```diff
@@ -7,12 +7,15 @@ import (
 	"github.com/multiformats/go-multiaddr"
 )
 
+//go:generate go run gen/main.go
+
 var ErrBadEncoding = errors.New("invalid message encoding")
 
 type Message struct {
 	Cid       cid.Cid
 	Addrs     [][]byte
 	ExtraData []byte `json:",omitempty"`
+	Republish bool
```

Comment:
Would it be possible to do this instead?
```go
type Message struct {                                                                                                                                                                                 
    Cid       cid.Cid                                                                                                                                                                                 
    Addrs     [][]byte                                                                                                                                                                                
    ExtraData []byte                                                                                                                                                                                  
    OrigPeer  string                                                                                                                                                                                  
}                                                                                                                                                                                                     
```
If `OrigPeer` != "" then it is a republish.  This way the original peer ID does not need to be encoded into `ExtraData`.

Reply:
Please edit at will; the commit by me was meant to show how to generate cbor :)